knowledge/pkg/datastore/documentloader/ocr/openai/openai.go

```diff
@@ -61,7 +63,8 @@ type RespMessage struct {
 
 type Choice struct {
 	FinishReason string      `json:"finish_reason"`
-	Message      RespMessage `json:"message"`
+	Message      RespMessage `json:"message"` // e.g. OpenAI provider resp
+	Delta        RespMessage `json:"delta"`   // e.g. Anthropic/Claude provider StreamResponse
```

Comment:
Should the logic be a seperate file like anthropic.go?

Reply:
This is going through the LLM Proxy anyway, so even OpenAI doesn't fit here anymore.
Also this theoretically applies for any model-provider we have or write that uses a stream response

Reply:
The point of the model providers is to make them such that the client doesn't need to know it's not OpenAI. If there is a difference between what the model provider produces and what you expect here, then the model provider should change.

Reply:
This will change completely with https://github.com/otto8-ai/otto8/issues/831 and https://github.com/otto8-ai/otto8/issues/832 - so I'd just leave it like this for now :+1:

Reply:
Follow-Up issue: https://github.com/obot-platform/obot/issues/831